specs/Standards/StorageManagement.md

```diff
@@ -0,0 +1,423 @@
+
+1. Alice queries `social::storage_balance_of({ "account_id": "alice" })`. With NEAR CLI:
+
+       near view social storage_balance_of '{"account_id": "alice"}'
+
+   Response:
+
+       {
+         total: '200000000000000000000000',
+         available: '100100000000000000000000'
+       }
+
+2. Alice calls `storage_withdraw` with a 1 yoctoⓃ deposit. NEAR CLI command:
+
+       near call social storage_withdraw \
+         '{"amount": "100100000000000000000000"}' \
+         --accountId alice --amount .000000000000000000000001
+
+   Result:
+
+       {
+         total: '200000000000000000000000',
+         available: '0'
+       }
+
+## Reference-level explanation
+
+**NOTES**:
+
+- All amounts, balances and allowance are limited by `U128` (max value 2<sup>128</sup> - 1).
+- This storage standard uses JSON for serialization of arguments and results.
+- Amounts in arguments and results are serialized as Base-10 strings, e.g. `"100"`. This is done to avoid JSON limitation of max integer value of 2<sup>53</sup>.
+- To prevent the deployed contract from being modified or deleted, it should not have any access keys on its account.
+
+**Interface**:
+
+```ts
+// The below defines the structure that will be returned for the methods:
+// * `storage_deposit`
+// * `storage_withdraw`
+// * `storage_balance_of`
+// The `total` and `available` values are string representations of unsigned
+// 128-bit integers.
+type StorageBalance = {
+   total: string;
+   available: string;
+}
+
+// The below structure will be returned for the method `storage_balance_bounds`.
+// Both `min` and `max` are string representations of unsigned 128-bit integers.
+//
+// `min` is the amount of tokens required to start using this contract at all
+// (eg to register with the contract). If a new contract user attaches `min`
+// NEAR to a `storage_deposit` call, subsequent calls to `storage_balance_of`
+// for this user must show their `total` equal to `min` and `available=0` .
+//
+// A contract may implement `max` equal to `min` if it only charges for initial
+// registration, and does not adjust per-user storage over time. A contract
+// which implements `max` must refund deposits that would increase a user's
+// storage balance beyond this amount.
+type StorageBalanceBounds = {
+    min: string;
+    max: string|null;
+}
+
+/************************************/
+/* CHANGE METHODS on fungible token */
+/************************************/
+// Payable method that receives an attached deposit of Ⓝ for a given account.
+//
+// If `account_id` is omitted, the deposit MUST go toward predecessor account.
+// If provided, deposit MUST go toward this account. If invalid, contract MUST
+// panic.
+//
+// If `registration_only=true`, contract MUST refund above the minimum balance
+// if the account wasn't registered and refund full deposit if already
+// registered.
+//
+// Any attached deposit in excess of `storage_balance_bounds.max` must be
+// refunded to predecessor account.
```

Comment:
the `current_storage_balance + attached_deposit` in excess of `storage_balance_bounds.max` ...

Reply:
updated to 

```
// The `storage_balance_of.total` + `attached_deposit` in excess of
// `storage_balance_bounds.max` must be refunded to predecessor account.
```